diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -2208,8 +2208,14 @@
       else
         req = get_option(opt_name)
         pkg_name = opts.get('pkg', name)
         if opts.has_key('versions')
-          foreach version: opts['versions']
-            dep = dependency(pkg_name, required: false, version: version, static: conf.get('static').is_odd())
+          _ver_found = false
+          foreach version: opts['versions']
+            if not _ver_found
+              dep = dependency(pkg_name, required: false, version: version, static: conf.get('static').is_odd())
+              if dep.found()
+                _ver_found = true
+              endif
+            endif
           endforeach
 
           if not dep.found() and req.enabled()
