diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -2581,7 +2581,10 @@
 if conf.get('zlib') == 1
   conf.set10('zlib_gzip', true)
-  gzip_exe = find_program('gzip', native: true)
+  gzip_exe = find_program('gzip', native: true, required: false)
   conf.set10('gzip', gzip_exe.found())
+  if not gzip_exe.found()
+    conf.set10('ptx_compression', false)
+  endif
 else
   conf.set10('ptx_compression', false)
 endif
